ros_ws/src/car_control/include/drive_parameters_multiplexer.h

```diff
@@ -0,0 +1,43 @@
+#pragma once
+
+#include <ros/ros.h>
+
+#include "car_control.h"
+#include "drive_parameters_source.h"
+#include <drive_msgs/drive_param.h>
+#include <std_msgs/Float64.h>
+#include <vector>
+
+/*
+*  This node subscribes to all publishers drive_param messages and selects one to forward to the car controller
+*/
+class DriveParametersMultiplexer
+{
+    public:
+    /**
+     * @brief Construct a new DriveParametersMultiplexer object and initialize sources for all 
+     * publishers of drive parameters
+     */
+    DriveParametersMultiplexer();
+    ~DriveParametersMultiplexer();
+
+    private:
+    ros::NodeHandle m_node_handle;
+
+    std::vector<DriveParametersSource*> m_sources;
```

Comment:
How would I do that?

Reply:
You could write 
```C++
std::vector<std::unique_ptr<DriveParametersSource>> m_sources;
```
Because of the ownership-semantic the self defined destructors becomes unnecessary.

The `new` in the constructor can then simply be replaced with `std::make_unique` e.g.
```C++
// Old
this->m_sources.push_back(
        new DriveParametersSource(&this->m_node_handle, "input/drive_param/wallfollowing", callback, 0, 0.1));
// With unique_ptr
this->m_sources.push_back(std::move(std::make_unique<DriveParametersSource>(&this->m_node_handle, "input/drive_param/wallfollowing", callback, 0, 0.1)))
```